xbmc/cores/VideoPlayer/VideoReferenceClock.cpp

```diff
@@ -108,7 +109,7 @@ void CVideoReferenceClock::Process()
       m_pVideoSync.reset();
     }
 
-    if (!SetupSuccess)
+    if (!SetupSuccess || m_destructing)
```

Comment:
Not sure the new member is needed. `CThread::StopThread` sets existing member `m_bStop` already. Maybe it can be used here instead of the new member? Or gets `StopThread` called to late by `CVideoReferenceClock::~CVideoReferenceClock`

Reply:
Yes, its called too late. After setting the stop event the thread (Process) continues running and setup of VideoSync is called before StopThread sets m_bStop. But Ok, I'll change back to m_bStop as it is not used in CThread internally and seems to be supposed for things like this.